### PR TITLE
ui/services: add integration key filter for services list

### LINKED ISSUE #2545: services: search by integration key
**What problem would you like to solve? Please describe:**
When debugging integrations with other systems, it may become necessary to lookup which service an integration key belongs to.

**Describe the solution you'd like:**
Add the ability to search on the Services list page with an integration key (matching what is copied from the integration key list page for all types).

**Describe alternatives you've considered:**
Connecting directly to the database and searching by key ID.

### PR DESCRIPTION
**Pull request by KatieMSB**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
This PR adds a new filter option for integration keys on the services list page.

**Which issue(s) this PR fixes:**
Fixes #2545 

**Screenshots:**
Screen Shot 2022-08-23 at 11 09 19 AM

**Describe any introduced API changes:**
Adds new search query for integration keys and updates existing services search to include new filter option.
```graphql
type IntegrationKeyConnection {
  nodes: [IntegrationKey!]!
  pageInfo: PageInfo!
}

input IntegrationKeySearchOptions {
  first: Int = 15
  after: String = ""
  search: String = ""
  omit: [String!]
}

# Allows searching for integration keys.
integrationKeys(input: IntegrationKeySearchOptions): IntegrationKeyConnection!
```

### CODE REVIEW

File: graphql2/graphqlapp/integrationkey.go [hunk 1]
```diff
@@ -58,3 +59,51 @@ func (key *IntegrationKey) Href(ctx context.Context, raw *integrationkey.Integra
...
+
+func (q *Query) IntegrationKeys(ctx context.Context, input *graphql2.IntegrationKeySearchOptions) (conn *graphql2.IntegrationKeyConnection, err error) {
+	if input == nil {
+		input = &graphql2.IntegrationKeySearchOptions{}
+	}
+
+	var opts integrationkey.InKeySearchOptions
+	if input.Search != nil {
+		opts.Search = *input.Search
+	}
```
Comment (tony-tvu) [inline graphql2/graphqlapp/integrationkey.go:71]: Currently, when there's no results for the search input, it returns a graphql error: 
graphql error

Suggested code change to return "No options" instead:
Screen Shot 2022-09-01 at 10 51 34 AM



```suggestion
	}
	conn = new(graphql2.IntegrationKeyConnection)
	err = validate.UUID("Search", opts.Search)
	if err != nil {
		conn.Nodes = []integrationkey.IntegrationKey{}
		return conn, nil
	}
 
```

**Review by tony-tvu — CHANGES_REQUESTED** [review]

File: graphql2/graphqlapp/integrationkey.go (hunk 1 above)
Comment (KatieMSB) [reply graphql2/graphqlapp/integrationkey.go:71]: Added pattern matching in the search query that will match the key id while typing and return 'No options' if there are no matches.

Comment (tony-tvu) [reply graphql2/graphqlapp/integrationkey.go:71]: :brain:

**Review by tony-tvu — APPROVED** [review]
LGTM! Also thanks for converting JS files to TS :smiley:

**Review by m17ch — APPROVED** [review]
LGTM 👍

**Pull request merged**